docs/proposals/policy-feg-api-cleanup/qos_enforcement.md

```diff
@@ -0,0 +1,333 @@
+
+## How We Will Change Magma
+
+### API Endpoints
+
+To enable configuration of QoS profiles and APNs, we will support the 
+following REST endpoints:
+
+```
+QoS Profiles
+
+GET     /lte/{network_id}/qos_profiles
+POST    /lte/{network_id}/qos_profiles
+
+GET     /lte/{network_id}/qos_profiles/{qos_name}
+PUT     /lte/{network_id}/qos_profiles/{qos_name}
+DELETE  /lte/{network_id}/qos_profiles/{qos_name}
+
+GET     /feg_lte/{network_id}/qos_profiles
+POST    /feg_lte/{network_id}/qos_profiles
+
+GET     /feg_lte/{network_id}/qos_profiles/{qos_name}
+PUT     /feg_lte/{network_id}/qos_profiles/{qos_name}
+DELETE  /feg_lte/{network_id}/qos_profiles/{qos_name}
+
+
+
+APN Profiles
+
+GET     /lte/{network_id}/apns
+POST    /lte/{network_id}/apns
+
+GET     /lte/{network_id}/apns/{apn_name}
+PUT     /lte/{network_id}/apns/{apn_name}
+DELETE  /lte/{network_id}/apns/{apn_name}
+
+GET     /feg_lte/{network_id}/apns
+POST    /feg_lte/{network_id}/apns
+
+GET     /feg_lte/{network_id}/apns/{apn_name}
+PUT     /feg_lte/{network_id}/apns/{apn_name}
+DELETE  /feg_lte/{network_id}/apns/{apn_name}
+
+
+
+APN Resources
+
+GET     /lte/{network_id}/gateways/{gateway_id}/resource_labels
+POST    /lte/{network_id}/gateways/{gateway_id}/resource_labels
+
+GET     /lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+PUT     /lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+DELETE  /lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+
+GET     /feg_lte/{network_id}/gateways/{gateway_id}/resource_labels
+POST    /feg_lte/{network_id}/gateways/{gateway_id}/resource_labels
+
+GET     /feg_lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+PUT     /feg_lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+DELETE  /feg_lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+```
+
+
+
+### API Entity Definitions
+
+Entity definitions for QoS, APN, and subscribers will change. 
+The final state is summarized here:
+
+```
+qos_profile:
+  id: string
+  class_id: enum[0,255]
+  max_req_bw_dl: uint32
+  max_req_bw_ul: uint32
+  # Guaranteed bit rate
+  # Not enforced by Magma yet
+  gbr:
+    gbr_ul: uint32
+    gbr_dl: uint32
+  # Allocation and retention priority
+  # Not enforced by Magma yet
+  arp:
+    arp_priority_level: uint32
+    arp_preemption_capability: bool
+    arp_preemption_vulnerability: bool
+ # Aggregated Maximum bit rate
+ # for non-GBR bearers, this limits the aggregate bit rate
+ # Enforce this on pipelined
+ apn_ambr:
+   apn_ambr_dl: uint32
+   apn_ambr_ul: uint32
+```
+
+As an initial goal, we would like to support apn_ambr functionality.
+
+```
+apn_configuration:
+  default_qos_profile: string
+  supported_qos_profiles: array[string]
+  # resource_label should match the name of a specified ApnResource
+  resource_label: string
+  # Unsupported as this time
+  # To be mandatory
+  pdn_type: enum[IPV4, IPV6, IPV4_IPV6]
```

Comment:
IPV4_IPv6 is not clear. Let's change this to either as IPV4V6 inline with `PDNType` definition in subscriberdb.proto or as IPV4_AND_IPV6.